src/serializer/ResidualFunctions.js

```diff
@@ -293,18 +296,32 @@ export class ResidualFunctions {
       let functionBody = t.blockStatement(rewrittenBody);
       let funcParams = params.slice();
       let funcOrClassNode;
-      let { classData } = instance;
 
-      if (classData !== undefined) {
-        let { methodType, classMethodKeyNode, classSuperNode, classMethodComputed } = classData;
+      if (this.residualClassMethodInstances.has(funcValue)) {
+        let classMethodInstance = this.residualClassMethodInstances.get(funcValue);
+        invariant(classMethodInstance);
+        let {
+          methodType,
+          classMethodKeyNode,
+          classSuperNode,
+          classMethodComputed,
+          classPrototype,
+          classMethodIsStatic,
+        } = classMethodInstance;
 
         let isConstructor = methodType === "constructor";
-        let homeObject = functionValue.$HomeObject;
-        invariant(homeObject instanceof ObjectValue);
+        invariant(classPrototype instanceof ObjectValue);
         invariant(classMethodKeyNode && (t.isExpression(classMethodKeyNode) || t.isIdentifier(classMethodKeyNode)));
-        // we use the $HomeObject as the key to get the class expression ast node
-        funcOrClassNode = this._getOrCreateClassNode(homeObject);
-        let classMethod = t.classMethod(methodType, classMethodKeyNode, funcParams, functionBody, classMethodComputed);
+        // we use the classPrototype as the key to get the class expression ast node
```

Comment:
Are there checks to ensure that the classPrototype uniquely corresponds to a single class expression AST node?